src/Domain/Creators/TimeDependentOptions/Python/BinaryCompactObject.cpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/TimeDependentOptions/Python/BinaryCompactObject.hpp"
+
+#include <array>
+#include <optional>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <vector>
+
+#include "Domain/Creators/TimeDependentOptions/BinaryCompactObject.hpp"
+#include "Domain/Creators/TimeDependentOptions/ExpansionMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/GridCenters.hpp"
+#include "Domain/Creators/TimeDependentOptions/RotationMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/ShapeMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/SkewMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/TranslationMap.hpp"
+#include "Domain/Structure/ObjectLabel.hpp"
+
+namespace py = pybind11;
+
+namespace domain::creators::time_dependent_options::py_bindings {
+void bind_binary_compact_object(py::module& m) {
+  py::class_<domain::creators::bco::TimeDependentMapOptions<false>>(
+      m, "BinaryCompactObjectTimeDependentOptions")
+      .def(py::init(
+               [](const double initial_time,
+                  const std::optional<domain::creators::time_dependent_options::
+                                    ExpansionMapOptions<false>>&
```

Comment:
Take these by value (not `const &`) and `std::move` below. Even better, use `py::init<...>()` listing all the types so you don't have to define the lambda at all. For the latter case you probably have to add `grid_centers = std::nullopt` as a default in the BinaryCompactObject TDO class or just bind grid centers TDO as well.